pkg/kademlia/lookup.go

```diff
@@ -38,60 +44,116 @@ func newSequentialLookup(rt *RoutingTable, nodes []*pb.Node, client node.Client,
 		heap.Init(&queue)
 	}
 
-	return &sequentialLookup{
+	return &lookup{
+		lMutex:          &sync.Mutex{},
 		contacted:       map[string]bool{},
 		queue:           queue,
 		slowestResponse: 0,
 		client:          client,
 		target:          target,
-		limit:           limit,
-		bootstrap:       bootstrap,
+		routines:        defaultRoutines,
+		rMutex:          &sync.Mutex{},
 	}
 }
 
-func (lookup *sequentialLookup) Run(ctx context.Context) error {
-	zero := &big.Int{}
-	targetBytes := new(big.Int).SetBytes(lookup.target.Bytes())
-
-	for len(lookup.queue) > 0 {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-		}
+func (lookup *lookup) Run(ctx context.Context) error {
```

Comment:
nit -- I would make it `l`

https://github.com/golang/go/wiki/CodeReviewComments#receiver-names